drake/systems/Simulation.h

```diff
@@ -74,11 +104,24 @@ void simulate(const System& sys, double ti, double tf,
       Eigen::VectorXd::Zero(getNumInputs(sys)));
   typename System::template OutputVector<double> y;
 
+  bool rt_warning_printed = false;
+
   // Take steps from ti to tf.
   double t = ti;
   while (t < tf) {
-    handle_realtime_factor(start, t, options.realtime_factor,
-                           options.timeout_seconds);
+    try {
+      handle_realtime_factor(start, t, options.realtime_factor,
+                             options.timeout_seconds);
+    } catch(std::runtime_error err) {
```

Comment:
Code that catches runtime_error (rather than a more specific exception type) can hide other errors -- this code may catch what it thinks is a realtime clock violation, but really could be any other type of error condition.  In other words the "argv[3]==warn" changes more than just timing errors into warnings, in turns all runtime_exceptions into warnings.